analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -60,9 +61,47 @@ protected override ProgramState PreProcess(ProgramState state, IBinaryOperationW
     {
         BinaryOperatorKind.Add => NumberConstraint.From(left.Min + right.Min, left.Max + right.Max),
         BinaryOperatorKind.Subtract => NumberConstraint.From(left.Min - right.Max, left.Max - right.Min),
+        BinaryOperatorKind.And when left.IsSingleValue && right.IsSingleValue => NumberConstraint.From(left.Min & right.Min),
+        BinaryOperatorKind.And => CalculateAnd(left, right),
         _ => null
     };
 
+    private static NumberConstraint CalculateAnd(NumberConstraint left, NumberConstraint right)
+    {
+        var min = left.Max < 0 && right.Max < 0
```

Comment:
I think `min` should be calculated from `Min`, not `Max`. Having
```
Left  Min      Max
      -64      -32
 11000000 11100000

Right Min      Max
      -63      - 2
 11000001 11111110
```
the zeros could turn themselves into ones, and could get killed. The lowest min bound is -64.